### ISSUE TITLE
[$15] Failed to mark notifications seen

### ISSUE THREAD

**Issue by maxceem** [appirio-tech/connect-app#2623]
From time to time there is error popup `Failed to mark notifications seen` when I navigate to the project dashboard.

![image](https://user-images.githubusercontent.com/146016/46844144-65561280-ce08-11e8-9501-9f4630f742cb.png)

**Comment by RishiRajSahu** [comment]
@maxceem as per me this appears when we try to mark already seen notifications as seen again. I tried reproducing it by opening the notifications dropdown in different browsers then trying to mark the notification as seen in each. For later it gave the same error as notification was marked as seen already.

**Comment by maxceem** [comment]
@RishiRajSahu as the user never calls the method to mark notifications as `seen` by himself, it's always done in the background automatically.

I think we can just remove showing this message in Connect App for this error. The only thing we would still have an error message in the browser console.

What do you think?

**Comment by maxceem** [comment]
As no PR was provided for 12 hours, this issue is now open for pick up.
The one who first comments can start working immediately without waiting to be assigned.

**Comment by maxceem** [comment]
Sure, assigned to keep it clear.

**Comment by gets0ul** [comment]
@maxceem 
Is there any specific step to do to trigger the popup?

**Comment by maxceem** [comment]
Yes, you will need two users, you can find credentials on the challenge forum.

- Use any user (but not `pshah_manager`) to create a new project, copy project URL and close the tab (important)
- Now use user `pshah_manger` from the forum and login in **another** browser 
- Using user `pshah_manger` open the project you've just created using link you copied before, go to the `Project Plan` tab, open any phase, go to the `Discussions` tab of the Phase and post a new comment. (you may need to reload page and make sure post is created, looks like there is some issue with creating posts)
- Copy URL of the current page, it will end with `/plan`
- Now using the first user in the first browser, open TWO tabs using the link which just copied end with `/plan`
- In on of the tabs, make sure you can see notification about a new post, open dropdown with notification in the top right corner, you should see "**Part1 Manag1**  responded to a post"
- In this tab click **Dashboard** tab and wait a few seconds, in the background this notification will be marked as seen when we open **Dashboard**
- Now switch to the second tab where you open link which ends with `/plan`. Here switch to the **Dashboard** tab too. It will try to mark the same notification as `seen` and background and you will see aleat popup with error

See the demo https://monosnap.com/file/xEH4yeZ2g3ExS9eQFN7EUfuv45sZem

Basically what is goning on we mark the same notification as seen two times and second one causes error. We call action `toggleNotificationSeen` to mark notifications as seen. As we don't have a direct button to call this action, and we only call it in the backgound, what we have to do, is not to show the error alert when error happens during marking notification as `seen`.

**Comment by maxceem** [comment]
@gets0ul please, let me know if you still willing to work on this issue after the clarifications.

**Comment by gets0ul** [comment]
@maxceem Yes, I'm working on it.

**Comment by gets0ul** [comment]
PR is done.

**Comment by maxceem** [comment]
@RishiRajSahu I guess this is already on PROD.

**Comment by RishiRajSahu** [comment]
@maxceem yep its in prod. Closing.
